c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -645,37 +659,55 @@ class DeviceCachingAllocator {
     return (p.block != nullptr);
   }
 
-  bool free_cached_blocks()
+  /** Free one oversize cached block of sufficient size **/
```

Comment:
I'd suggest updating the comment here to clarify that this function releases an oversize cached block to the system allocator, to clarify to the reader that `free` is when a block is returned to the caching allocator, whereas `release` is when a block is returned to the system allocator.